extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/query/StatisticComputingIterator.java

```diff
@@ -0,0 +1,268 @@
+			return getTopStatValue();
+		}
+		return null;
+	}
+
+	@Override
+	public boolean hasTop() {
+		// firstly iterate through all of the original data values
+		final boolean hasTopOriginal = hasTopOriginal();
+		if (hasTopOriginal) {
+			return true;
+		}
+		return hasTopStat();
+	}
+
+	@Override
+	protected boolean accept(
+			final Text currentRow,
+			final List<Key> keys,
+			final List<Value> values,
+			final CommonIndexModel model,
+			final CommonIndexedPersistenceEncoding persistenceEncoding ) {
+		if (super.accept(
+				currentRow,
+				keys,
+				values,
+				model,
+				persistenceEncoding)) {
+			if (persistenceEncoding.getAdapterId().getString().equals(
+					adapter.getAdapterId().getString())) {
+				final PersistentDataset<Object> adapterExtendedValues = new PersistentDataset<Object>();
+				if (persistenceEncoding instanceof IndexedAdapterPersistenceEncoding) {
+					final PersistentDataset<Object> existingExtValues = ((IndexedAdapterPersistenceEncoding) persistenceEncoding).getAdapterExtendedData();
+					if (existingExtValues != null) {
+						for (final PersistentValue<Object> val : existingExtValues.getValues()) {
+							adapterExtendedValues.addValue(val);
+						}
+					}
+				}
+				final PersistentDataset<byte[]> stillUnknownValues = new PersistentDataset<byte[]>();
+				final List<PersistentValue<byte[]>> unknownDataValues = persistenceEncoding.getUnknownData().getValues();
+				for (final PersistentValue<byte[]> v : unknownDataValues) {
+					final FieldReader<Object> reader = adapter.getReader(v.getId());
+					final Object value = reader.readField(v.getValue());
+					adapterExtendedValues.addValue(new PersistentValue<Object>(
+							v.getId(),
+							value));
+				}
+				if (persistenceEncoding instanceof IndexedAdapterPersistenceEncoding) {
+					for (final PersistentValue<Object> v : ((IndexedAdapterPersistenceEncoding) persistenceEncoding).getAdapterExtendedData().getValues()) {
+						adapterExtendedValues.addValue(v);
+					}
+				}
+				final IndexedAdapterPersistenceEncoding encoding = new IndexedAdapterPersistenceEncoding(
+						persistenceEncoding.getAdapterId(),
+						persistenceEncoding.getDataId(),
+						persistenceEncoding.getIndexInsertionId(),
+						persistenceEncoding.getDuplicateCount(),
+						persistenceEncoding.getCommonData(),
+						stillUnknownValues,
+						adapterExtendedValues);
+				final Object row = adapter.decode(
+						encoding,
+						new PrimaryIndex(
+								null, // the data adapter can't use the numeric
+										// index
+										// strategy and only the common index
+										// model to decode which is the case for
+										// feature data,
+										// we pass along a null strategy to
+										// eliminate the necessity to send a
+										// serialization of the strategy in the
+										// options of this iterator
+								model));
+				if (row != null) {
+					final List<ByteArrayId> insertionIds = new ArrayList<ByteArrayId>();
```

Comment:
this is just needed to satisfy the datastatistics interface.  Duplicates will be interesting for this interface, but ultimately I'd like this to be a different interface - call it aggregations or something
but you do want to be able to reuse existing datastatistics so they'd all have to implement an aggregation interface or whatever this is called